### ISSUE TITLE
Make canvas.drawingBuffer optional

### ISSUE THREAD

**Issue by kevinpapst** [ericblade/quagga2#151]
As discussed in #140 the `canvas.drawingBuffer` seems not to be required during runtime, as least in cases where you only want to detect barcodes.

From a first look through the code there is no real need for this element, which currently has to be hidden via CSS. And it doesn't hurt if the element is removed from the DOM via JS after initialization.

Some proposed changes, which could work:
- use it ONLY if the element exists
- if it doesn't exist: ignore it and do NOT create it from within the library
- add the canvas manually in the examples
- add some documentation about its usage to the debug section
- document this possible BC break in the release

What do you think @ericblade ? Does that make sense? 
It somehow collides with what you wrote in #33 - in that discussion it sounds as if the element is required.

**Comment by ericblade** [comment]
yeah, that actually sounds preferable.  Make the user control it, instead of having people wondering why they have to apply a bunch of CSS to make drawing work.

I remember there's at least one or two references to the drawingBuffer where some properties are assigned to it (one of them was during window resize, i think) so those sections might need to be guarded to prevent assigning into an undefined...

i think this would replace #33, probably.

**Comment by ericblade** [comment]
Hey @kevinpapst in more recent changes, some of the relevant code has been moved over to src/quagga/initCanvas.ts .  

If you'd like to do a pull req on this, we should just need to make sure any accesses to canvas.overlay are properly guarded, update any samples in the repo, and make sure the documentation points out that if you want to draw on top of it, we'll provide the interface, if you provide the #drawingBuffer .

**Comment by ericblade** [comment]
@kevinpapst if you're still out here... I've got Copilot doing a bunch of analysis on this topic right now, it's handling some documentation here

https://github.com/ericblade/quagga2/commit/49c3aff4fd6a6ad94e37aeff0b1f91a5b4a660c1

seems that it's quite a handy feature, but as this topic does say, it might well be possible to make it optional.  I'll be adding that to analysis soon, looks like it could probably be added as a configuration option.  I can see benefit to being able to turn off any canvases that aren't needed

**Comment by ericblade** [comment]
added configuration element 
```
canvas { createOverlay: bool }
```
defaults true

**Comment by kevinpapst** [comment]
I am around 😁 and I am almost certain that this is one of the longest waiting feature requests that were really solved at the end.

Thanks. Will give it a try, once it's released.

**Comment by ericblade** [comment]
haha. right...  Thanks for being around still!  I've been throwing a lot of tasks at Copilot lately, and picking off some others, myself, hope to validate them all this weekend or next in my own app, and release then.

**Comment by kevinpapst** [comment]
Was this released? I don't see it mentioned in 1.11.0

**Comment by ericblade** [comment]
> Was this released? I don't see it mentioned in 1.11.0

Should be! top of the release changelog https://github.com/ericblade/quagga2/releases

I haven't actually tested what it looks like in the DOM when you use it, only that it doesn't go through the creation step, and shouldn't crash anything if you disable it.   

should disable the built-in overlay canvas if you add ``` canvas: { createOverlay: false } ``` to your config object.  Defaults to true so as not to break any existing installations

**Comment by kevinpapst** [comment]
Ah thanks, I looked for "drawing" and "buffer" and the issue number, but couldn't find any of that. 
Will give it a try and only report back, if there were problems.

Thanks for taking care of that one 👍

**Comment by ericblade** [comment]
hmm. makes sense.  yeah i guess i didn't think that through too much, was just trying to get through the backlog.  i'll make a note to make that make more sense naming wise for a 2.0
